device/tt_device.cpp

```diff
@@ -27,4 +27,6 @@ tt_device::tt_device(const std::string& sdesc_path) : soc_descriptor_per_chip({}
 tt_device::~tt_device() {
 }
 
-const tt_SocDescriptor *tt_device::get_soc_descriptor(chip_id_t chip) const { return &soc_descriptor_per_chip.at(chip); }
+tt_SocDescriptor& tt_device::get_soc_descriptor(chip_id_t chip_id){
```

Comment:
Can this be `const tt_SocDescriptor&`?  No `const` here implies that the caller can mutate it.  (I don't know if mutating it is necessary... ideally the SOC doesn't change?)